internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -909,7 +909,7 @@ func selectClusterNetworkType(params *models.V2ClusterUpdateParams, cluster *com
 			return false
 		}
 		return network.IsIPv6CIDR(*ip)
-	})) {
+	})) || common.IsSingleNodeCluster(cluster) {
```

Comment:
Seems strange that this is the only place this change is required, what about the REST API?

Reply:
@eranco74 Doesn't the validator https://github.com/openshift/assisted-service/pull/3419/files/965c7c9a42d13b3f501bb9392572b4d2cfd26c02#diff-a712b20e2e375b0286d437906cfa73f4423a506449c19e71ba14633cd89dd01b serve the REST API? According to the unit tests it does. In any case, it's still a WIP, I will keep looking for other places.

Reply:
So after going over the code it seems that if no network type was provided by the user the assisted-installer defaults to the  network type returned by the Openshift installer (decided when generating the install config) this means that different OCP versions will return different network type

Reply:
@eranco74 @romfreiman IIUC it means an SNO installation may end up having OpenShiftSDN, which is exactly what we want to avoid. Do you think we should default to OVNKubernetes in the API, too? Or make network type mandatory in case of SNO?

Reply:
I think we should default in the API.
While there is a similar task for the OpenShift installer, I don't think we want to depend on it or on the version it will get delivered to.
Adding a default in the assisted-service API also adds visibility of the network type prior to the installation.